### PR TITLE
fix: Align resource conversion for empty values

### PR DESCRIPTION
**Pull request by jeromelaban**
GitHub Issue (If applicable): closes https://github.com/unoplatform/uno/issues/8800

## PR Type

What kind of change does this PR introduce?
- Bugfix

## What is the new behavior?

x:Uid assigned string empty values are now returning null to align with the WinUI behavior.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI.Tests/ResourceLoader/Strings/en/Resources.Designer.cs [hunk 1]
```diff
@@ -61,29 +61,29 @@ internal class Resources {
...
+        internal static string TestEmptyConverterText_ValueIfNull {
             get {
-                return ResourceManager.GetString("Given_ResourceLoader.When_LocalizedResource", resourceCulture);
+                return ResourceManager.GetString("TestEmptyConverterText.ValueIfNull", resourceCulture);
             }
         }
         
         /// <summary>
-        ///   Looks up a localized string similar to Header in &apos;en&apos;.
+        ///   Looks up a localized string similar to .
```
Comment (agneszitte-nventive) [inline src/Uno.UI.Tests/ResourceLoader/Strings/en/Resources.Designer.cs:82]: similar to ?

File: src/Uno.UI.Tests/ResourceLoader/Strings/en/Resources.Designer.cs [hunk 2]
```diff
@@ -61,29 +61,29 @@ internal class Resources {
         }
         
         /// <summary>
-        ///   Looks up a localized string similar to App70-en.
+        ///   Looks up a localized string similar to .
```
Comment (agneszitte-nventive) [inline src/Uno.UI.Tests/ResourceLoader/Strings/en/Resources.Designer.cs:64]: Similar to ?

**Review by agneszitte-nventive — APPROVED** [review]

File: src/Uno.UI.Tests/ResourceLoader/Strings/en/Resources.Designer.cs (hunk 1 above)
Comment (jeromelaban) [reply src/Uno.UI.Tests/ResourceLoader/Strings/en/Resources.Designer.cs:82]: Those are generated files and the ordering has changed. We can ignore this.

File: src/Uno.UI.Tests/ResourceLoader/Strings/en/Resources.Designer.cs (hunk 2 above)
Comment (jeromelaban) [reply src/Uno.UI.Tests/ResourceLoader/Strings/en/Resources.Designer.cs:64]: Those are generated files and the ordering has changed. We can ignore this.

**Review by carldebilly — APPROVED** [review]

**Review by ajpinedam — APPROVED** [review]

**Pull request merged**